src/lib/zod/tags.ts

```diff
@@ -0,0 +1,10 @@
+import z from 'zod';
+
+export const tagSchema = z.object({
+  name: z
+    .string()
+    .min(1, 'A tag name is required')
+    .max(50, 'The maximum length for a tag is 50 characters'),
+});
```

Comment:
```suggestion
 name: z
    .string()
    .refine(
      (value) => value.trim() !== '',
      { message: 'A tag name is required' }
    )
    .refine(
      (value) => value.trim().length <= 50,
      { message: 'The maximum length for a tag is 50 characters' }
    )
    .refine(
      (value) => !/\s{2,}/.test(value),
      { message: 'Consecutive spaces are not allowed' }
    ),
});
```

This should, at least to my knowledge, work on Zod.  Should do the same as the regex on TagForm component.

Reply:
Yes, definitely worth implementing!